src/printlayoutlistmodel.cpp

```diff
@@ -0,0 +1,76 @@
+/***************************************************************************
+  printlayoutlistmodel.cpp
+ ---------------------
+ begin                : 20.8.2018
+ copyright            : (C) 2018 by david
+ email                : david at opengisch dot ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include "printlayoutlistmodel.h"
+#include <qgsproject.h>
+
+PrintLayoutListModel::PrintLayoutListModel(QObject* parent)
+  : QAbstractListModel( parent )
+{
+}
+
+QHash<int, QByteArray> PrintLayoutListModel::roleNames() const
+{
+  QHash<int, QByteArray> roles = QAbstractListModel::roleNames();
+  roles[TitleRow]  = "Title";
+  roles[IndexRow] = "Index";
+
+  return roles;
+}
+
+QgsProject* PrintLayoutListModel::project() const
+{
+  return mProject;
+}
+
+void PrintLayoutListModel::setProject( QgsProject* project )
+{
+  mProject = project;
```

Comment:
Better add a guard here

```if ( mProject == project ) 
return;
?```

Reply:
Because it's used for refreshing model as well, I would prefer:
```
  if( mProject != project )
    mProject = project;

  reloadModel();
}
```
What do you think?

Reply:
Why not call `reloadModel()` explicitly in such cases?

Reply:
I thought because it needs to reload after loading project anyway. But I can do that and then there are two calls onLoadProjectEnded. No problem.